src/components/widget/component.js

```diff
@@ -115,21 +97,43 @@ class Widget extends PureComponent {
           ? <Spinner isLoading />
           : (
             <div className={classnames(styles.content)}>
-              {children({
-                name,
-                isActive,
-                isCollapsed,
-                isLoading,
-                layersIds,
-                slug,
-                data: this.getDataBySlug(slug),
-                ...props,
-              })}
+              <Template {...templateProps} />
             </div>
           )
         }
       </div>
     );
   }
 }
+
+Widget.propTypes = {
+  data: PropTypes.shape({}),
+  highlightedPlaces: PropTypes.arrayOf(PropTypes.shape({})),
+  name: PropTypes.string.isRequired,
+  slug: PropTypes.string.isRequired,
+  widgetConfig: PropTypes.shape({}).isRequired,
+  layerId: PropTypes.string,
+  layersIds: PropTypes.arrayOf(PropTypes.string),
+  location: PropTypes.shape({}),
+  isActive: PropTypes.bool,
+  isCollapsed: PropTypes.bool,
+  isLoading: PropTypes.bool,
+  toggleActive: PropTypes.func,
+  toggleCollapse: PropTypes.func,
+  template: PropTypes.func.isRequired
+};
+
+Widget.defaultProps = {
+  data: null,
+  highlightedPlaces: null,
+  isActive: false,
+  isCollapsed: false,
+  isLoading: false,
+  layerId: null,
+  layersIds: null,
+  location: null,
+  toggleActive: () => { },
+  toggleCollapse: () => { }
```

Comment:
I think this props should go back to static so we make sure we don´t declare them again when re render